src/Database/PostgreSQL/PQTypes/Model/Trigger.hs

```diff
@@ -0,0 +1,284 @@
+    <+> "EXECUTE FUNCTION" <+> trgFunction
+    <+> "();"
+  where
+    trgName
+      | triggerName == "" = error "Trigger must have a name."
+      | otherwise = triggerMakeName triggerName triggerTable
+    trgEvents
+      | triggerEvents == Set.empty = error "Trigger must have at least one event."
+      | otherwise = mintercalate " OR " . map triggerEventName $ Set.toList triggerEvents
+    trgTiming = let deferrable = (if triggerDeferrable then "" else "NOT") <+> "DEFERRABLE"
+                    deferred   = if triggerInitiallyDeferred
+                                 then "INITIALLY DEFERRED"
+                                 else "INITIALLY IMMEDIATE"
+                in deferrable <+> deferred
+    trgWhen = maybe "" ("WHEN" <+>) triggerWhen
+    trgFunction = tfName triggerFunction
+
+-- | Get all noninternal triggers from the database.
+--
+-- Run a query that returns all database triggers marked as @tgisinternal = false@.
+--
+-- Note that, in the background, to get the trigger's @WHEN@ clause and the source code of
+-- the attached function, the entire query that had created the trigger is received using
+-- @pg_get_triggerdef(t.oid)::text@ and then parsed. The result of that call will be
+-- decompiled and normalized, which means that it's likely not what the user had
+-- originally actually typed. Therefore, 'triggerWhen' and 'tfSource' of 'triggerFunction'
+-- should not be relied upon when comparing the original 'Trigger' that had been used for
+-- creating in the database and the one received by this function.
+--
+-- @since 1.15.0
+getDBTriggers :: forall m. MonadDB m => m [Trigger]
+getDBTriggers = do
+  runQuery_ . sqlSelect "pg_trigger t" $ do
+    sqlResult "t.tgname::text" -- name
+    sqlResult "t.tgtype" -- smallint == int2 => (2 bytes)
+    sqlResult "t.tgdeferrable" -- boolean
+    sqlResult "t.tginitdeferred"-- boolean
+    -- This gets the entire query that created this trigger. Note that it's decompiled and
+    -- normalized, which means that it's likely not what the user actually typed. For
+    -- example, if the original query had excessive whitespace in it, it won't be in this
+    -- result.
+    -- TODO: Do we want to remove one layer of () similarly to how Checks.hs does that?
+    --sqlResult "regexp_replace(pg_get_triggerdef(t.oid)::text, 'WHEN \\((.*)\\)', 'WHEN \\1')"
+    sqlResult "pg_get_triggerdef(t.oid)::text"
+    sqlResult "p.proname::text" -- name
+    sqlResult "p.prosrc" -- text
+    sqlResult "c.relname::text"
+    sqlJoinOn "pg_proc p" "t.tgfoid = p.oid"
+    sqlJoinOn "pg_class c" "c.oid = t.tgrelid"
+    sqlWhereEq "t.tgisinternal" False
+  fetchMany getTrigger
+  where
+    getTrigger :: (String, Int16, Bool, Bool, String, String, String, String) -> Trigger
+    getTrigger (tgname, tgtype, tgdeferrable, tginitdeferrable, triggerdef, proname, prosrc, tblName) =
+      Trigger { triggerTable = tableName
+              , triggerName = triggerBaseName (fromString tgname) tableName
```

Comment:
Are you sure? `tgname` is a String and using `fromString` reads well. If I change it, it will read `rawSQL (Text.pack tgname) ()` which may be explicit, but much harder to read.

Reply:
It's about `unsafeSQL` which is just a restricted version of `fromString`, not `rawSQL`.